src/DependencyInjection/HttplugExtension.php

```diff
@@ -292,6 +294,24 @@ private function configurePluginByName($name, Definition $definition, array $con
 
                 break;
 
+            case 'throttle':
+                if (!\class_exists(ThrottlePlugin::class)) {
+                    throw new InvalidConfigurationException('You need to require the Throttle Plugin to be able to use it: "composer require php-http/throttle-plugin".');
+                }
+
+                $key = $config['name'] ? '.'.$config['name'] : '';
```

Comment:
if name is required, this check is for the special case where name is set to `''`? below it would still do strange things when name is `''`

Reply:
and i think symfony configuration will not allow `''` on something that is set to be required

Reply:
Done, check #462